eclipse-collections-api/src/main/java/org/eclipse/collections/api/factory/bag/ImmutableBagFactory.java

```diff
@@ -47,8 +49,50 @@
      */
     <T> ImmutableBag<T> of(T... elements);
 
+    default <T> ImmutableBag<T> ofOccurrences(T element, int occurrence)
+    {
+        return this.withOccurrences(element, occurrence);
+    }
+
+    default <T> ImmutableBag<T> ofOccurrences(T element1, int occurrence1, T element2, int occurrence2)
+    {
+        return this.withOccurrences(element1, occurrence1, element2, occurrence2);
+    }
+
+    default <T> ImmutableBag<T> ofOccurrences(T element1, int occurrence1, T element2, int occurrence2, T element3, int occurrence3)
+    {
+        return this.withOccurrences(element1, occurrence1, element2, occurrence2, element3, occurrence3);
+    }
+
+    default <T> ImmutableBag<T> ofOccurrences(T element1, int occurrence1, T element2, int occurrence2, T element3, int occurrence3, T element4, int occurrence4)
+    {
+        return this.withOccurrences(element1, occurrence1, element2, occurrence2, element3, occurrence3, element4, occurrence4);
+    }
+
+    default <T> ImmutableBag<T> ofOccurrences(ObjectIntPair<T>... elementsWithOccurrences)
+    {
+        return this.withOccurrences(elementsWithOccurrences);
+    }
+
+    default ImmutableBag<Integer> ofOccurrences(IntIntPair... elementsWithOccurrences)
+    {
+        return this.withOccurrences(elementsWithOccurrences);
+    }
+
     <T> ImmutableBag<T> with(T... elements);
 
+    <T> ImmutableBag<T> withOccurrences(T element, int occurrence);
+
+    <T> ImmutableBag<T> withOccurrences(T element1, int occurrence1, T element2, int occurrence2);
+
+    <T> ImmutableBag<T> withOccurrences(T element1, int occurrence1, T element2, int occurrence2, T element3, int occurrence3);
+
+    <T> ImmutableBag<T> withOccurrences(T element1, int occurrence1, T element2, int occurrence2, T element3, int occurrence3, T element4, int occurrence4);
+
+    <T> ImmutableBag<T> withOccurrences(ObjectIntPair<T>... elementsWithOccurrences);
+
+    ImmutableBag<Integer> withOccurrences(IntIntPair... elementsWithOccurrences);
```

Comment:
I do not understand why `withOccurrences(IntIntPair... elementsWithOccurrences) `is necessary.  If there is a use case for this, I would rather add the factory methods to the primitive factories, so this returns an `ImmutableIntBag`.

I understand why you might want to replace this with `ObjectIntPair` here to return an `ImmutableBag<T>`, but `IntIntPair` returning a boxed `ImmutableBag<Integer>` does not make sense.

Reply:
> `withOccurrences(IntIntPair... elementsWithOccurrences)`

The reason I provided this API is I already provided the following API:
```java
withOccurrences(ObjectIntPair<T>... elementsWithOccurrences);
```

Above API will need to use:
```java
PrimitiveTuples.pair("2", 2)
```

Which will return `ObjectIntPair`.
If I use this:
```java
PrimitiveTuples.pair(2, 2)
```

Which will return `IntIntPair`.
That's why I have `withOccurrences(IntIntPair... elementsWithOccurrences);`


> IntIntPair returning a boxed ImmutableBag<Integer> doesn't make sense to me.


If the APIs with varargs make sense to you, I could see the possibility of returning `ImmutableIntBag` instead of the current one. If there is a better collection than `ImmutableIntBag`, please let me know too. Thanks.

Reply:
This method would make more sense on `IntBags` in the `MutableIntBagFactory` and `ImmutableIntBagFactory`.

For Example:
```
IntBag bag = IntBags.mutable.withOccurrences(
    PrimitiveTuples.pair(1, 10), 
    PrimitiveTuples.pair(2, 20));
```
There is also `ShortBag`, `ByteBag`, `FloatBag`, `LongBag`, `DoubleBag`, `BooleanBag`.  They should have the equivalent methods.  But I would recommend doing this in a separate PR.

Reply:
Any update on this @jiminhsieh?

Reply:
@donraab I don't have any update yet, I was busy at the end of last year. However, this PR will be the first finished PR for 2020. :)